src/Framework/AspNetCoreAnalyzers/test/WebApplicationBuilder/PublicPartialProgramClassTest.cs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis.Testing;
+using VerifyCS = Microsoft.AspNetCore.Analyzers.Verifiers.CSharpCodeFixVerifier<
+    Microsoft.AspNetCore.Analyzers.PublicPartialProgramClassAnalyzer,
+    Microsoft.AspNetCore.Fixers.PublicPartialProgramClassFixer>;
+
+namespace Microsoft.AspNetCore.Analyzers;
+
+public class PublicPartialProgramClassTest
+{
+    [Fact]
+    public async Task DoesNothingWhenNoDeclarationIsFound()
+    {
+        // Arrange
+        var source = """
+using Microsoft.AspNetCore.Builder;
+
+var app = WebApplication.Create();
+
+app.MapGet("/", () => "Hello, World!");
+
+app.Run();
+""";
+
+        // Assert
+        await VerifyCS.VerifyCodeFixAsync(source, source);
+    }
+
+    [Fact]
+    public async Task RemovesDeclarationIfItIsFound()
+    {
+        // Arrange
+        var source = """
+using Microsoft.AspNetCore.Builder;
+
+var app = WebApplication.Create();
+
+app.MapGet("/", () => "Hello, World!");
+
+app.Run();
+
+{|#0:public partial class Program { }|}
```

Comment:
Two test case suggestions in case they would appear as "not empty":

1. This, but with a semicolon instead of {} (`Program;`) if that actually usually compiles;
1. This, but with a single comment between the braces (because I've done this with the comment explaining why the declaration is there: [example](https://github.com/martincostello/dotnet-minimal-api-integration-testing/blob/95da381be986d3820048846faf646219687ae2ee/src/TodoApp/Program.cs#L74)).

Reply:
Good recs -- added a few more tests here. Let me know what you think?